tsl/src/remote/dist_txn.h

```diff
@@ -18,8 +18,34 @@ extern TSConnection *remote_dist_txn_get_connection(TSConnectionId id,
 													RemoteTxnPrepStmtOption prep_stmt);
 
 #ifdef DEBUG
-extern void (*testing_callback_call_hook)(const char *event);
-#endif
+
+typedef enum DistTransactionEvent
+{
+	DTXN_EVENT_ANY,
+	DTXN_EVENT_PRE_COMMIT,
+	DTXN_EVENT_WAIT_COMMIT,
+	DTXN_EVENT_PRE_ABORT,
+	DTXN_EVENT_PRE_PREPARE,
+	DTXN_EVENT_WAIT_PREPARE,
+	DTXN_EVENT_POST_PREPARE,
+	DTXN_EVENT_PRE_COMMIT_PREPARED,
+	DTXN_EVENT_WAIT_COMMIT_PREPARED,
+	DTXN_EVENT_SUB_XACT_ABORT,
+} DistTransactionEvent;
+
+#define MAX_DTXN_EVENT (DTXN_EVENT_SUB_XACT_ABORT + 1)
```

Comment:
You dont like place MAX into the same enum in order to avoid warning messages in case if it is not used in an swith?

Reply:
the MAX value would not be a valid event.